cranelift/codegen/src/isa/arm64/mod.rs

```diff
@@ -1,6 +1,219 @@
+//! ARM 64-bit Instruction Set Architecture.
+
+use crate::ir::Function;
+use crate::isa::Builder as IsaBuilder;
+use crate::isa::TargetIsa;
+use crate::machinst::{
+    compile, MachBackend, MachCompileResult, ShowWithRRU, TargetIsaAdapter, VCode,
+};
+use crate::result::CodegenResult;
+use crate::settings;
+
+use alloc::boxed::Box;
+use std::str::FromStr;
+
+use regalloc::RealRegUniverse;
+use target_lexicon::Triple;
+
+// New backend:
 mod abi;
 mod inst;
 mod lower;
 
-/// Placeholder for later implementation.
-pub struct Arm64Backend {}
+use inst::create_reg_universe;
+
+/// An ARM64 backend.
+pub struct Arm64Backend {
+    flags: settings::Flags,
+}
+
+impl Arm64Backend {
+    /// Create a new ARM64 backend with the given (shared) flags.
+    pub fn new_with_flags(flags: settings::Flags) -> Arm64Backend {
+        Arm64Backend { flags }
+    }
+
+    fn compile_vcode(&self, mut func: Function, flags: &settings::Flags) -> VCode<inst::Inst> {
+        // This performs lowering to VCode, register-allocates the code, computes
+        // block layout and finalizes branches. The result is ready for binary emission.
+        let abi = Box::new(abi::ARM64ABIBody::new(&func));
+        compile::compile::<Arm64Backend>(&mut func, self, abi, flags)
+    }
+}
+
+impl MachBackend for Arm64Backend {
+    fn compile_function(
+        &self,
+        func: Function,
+        want_disasm: bool,
+    ) -> CodegenResult<MachCompileResult> {
+        let flags = self.flags();
+        let vcode = self.compile_vcode(func, flags);
+        let sections = vcode.emit();
+        let frame_size = vcode.frame_size();
+
+        let disasm = if want_disasm {
+            Some(vcode.show_rru(Some(&create_reg_universe())))
+        } else {
+            None
+        };
+
+        Ok(MachCompileResult {
+            sections,
+            frame_size,
+            disasm,
+        })
+    }
+
+    fn name(&self) -> &'static str {
+        "arm64"
+    }
+
+    fn triple(&self) -> Triple {
+        FromStr::from_str("arm64").unwrap()
+    }
+
+    fn flags(&self) -> &settings::Flags {
+        &self.flags
+    }
+
+    fn reg_universe(&self) -> RealRegUniverse {
+        create_reg_universe()
+    }
+}
+
+/// Create a new `isa::Builder`.
+pub fn isa_builder(triple: Triple) -> IsaBuilder {
+    IsaBuilder {
+        triple,
+        setup: settings::builder(),
+        constructor: isa_constructor,
```

Comment:
```suggestion
        constructor: |_, shared_flags, _| {
            let backend = Arm64Backend::new_with_flags(shared_flags);
            Box::new(TargetIsaAdapter::new(backend))
        },
```

If a closure doesn't capture any variables, it can coerce to a function pointer.